docs/cookbook/rag.md

```diff
@@ -1,7 +1,424 @@
-# RAG (Retrieval-Augmented Generation)
+# Retrieval Augmented Generation (RAG)
+
+When we need to provide additional information to the model that it hasn't yet been trained on, we can retrieve the relevant information from an external source and provide it as context to our model. For our example, we will use a dataset of [BBC news articles from 2004](https://raw.githubusercontent.com/Dawit-1621/BBC-News-Classification/main/Data/BBC%20News%20Test.csv). We will:
+
+- query the dataset with a topic of our choosing
+- retrieve a number of relevant articles
+- ask our chat model to summarize the relevant ones
+- show how to query the dataset both locally and using a vector database [(Pinecone)](https://docs.pinecone.io/)
+- show how Mirascope can simplify RAG.
+
+!!! note
+
+    The following code snippets have been moved around for the sake of clarity in the walkthrough, and may not work if you copy and paste them. For a fully functional script, take a look at the [code](https://github.com/Mirascope/mirascope/blob/main/cookbook/rag_examples/) in our repo.
+
+## Before we get started
+
+If you haven't already, it will be worth taking a look some of our relevant concept pages for a more detailed explanation of the functionality we'll be using in this walkthrough.
+
+- [Pydantic Prompts](https://docs.mirascope.io/latest/concepts/pydantic_prompts/) (template variables, messages)
+- [LLM Convenience Wrappers](https://docs.mirascope.io/latest/concepts/llm_convenience_wrappers/#openaichat)
+
+In addition, here are the variables we will be using in this cookbook recipe:
+
+```python
+# config.py
+
+from typing import Optional
+
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+MODEL = "gpt-3.5-turbo"
+EMBEDDINGS_MODEL = "text-embedding-ada-002"
+MAX_TOKENS = 1000
+TEXT_COLUMN = "Text"
+EMBEDDINGS_COLUMN = "embeddings"
+FILENAME = "news_article_dataset.pkl"
+URL = "https://raw.githubusercontent.com/Dawit-1621/BBC-News-Classification/main/Data/BBC%20News%20Test.csv"
+PINECONE_INDEX = "news-articles"
+PINECONE_NAMESPACE = "articles"
+
+class Settings(BaseSettings):
+    openai_api_key: Optional[str] = None
+    pinecone_api_key: Optional[str] = None
+
+    model_config = SettingsConfigDict(env_file=".env")
+```
+
+```python
```

Comment:
nit: switch the .env to be above the config. Realizing it makes more sense to show those first and then the Settings object in config.py